app/screens/reschedule_draft/reschedule_draft.tsx

```diff
@@ -0,0 +1,167 @@
+
+type Props = {
+    currentUserTimezone?: UserTimezone | null;
+    componentId: AvailableScreens;
+    closeButtonId: string;
+    draft: ScheduledPostModel;
+}
+
+const OPTIONS_PADDING = 12;
+
+const styles = StyleSheet.create({
+    body: {
+        flex: 1,
+    },
+    container: {
+        flex: 1,
+    },
+    loader: {
+        flex: 1,
+        alignItems: 'center',
+        justifyContent: 'center',
+    },
+    optionsContainer: {
+        paddingTop: OPTIONS_PADDING,
+    },
+});
+
+const RIGHT_BUTTON = buildNavigationButton('reschedule-draft', 'reschedule_draft.save.button');
+
+const RescheduledDraft: React.FC<Props> = ({
+    currentUserTimezone,
+    componentId,
+    closeButtonId,
+    draft,
+}) => {
+    const theme = useTheme();
+    const intl = useIntl();
+    const serverUrl = useServerUrl();
+    const {database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+    const mainView = useRef<View>(null);
+    const [isUpdating, setIsUpdating] = useState(false);
+    const [errorLine, setErrorLine] = useState<string | undefined>();
+    const [errorExtra, setErrorExtra] = useState<string | undefined>();
+    const selectedTime = useRef<string | null>(null);
+    const userTimezone = getTimezone(currentUserTimezone);
+
+    const toggleSaveButton = useCallback((enabled = false) => {
+        setButtons(componentId, {
+            rightButtons: [{
+                ...RIGHT_BUTTON,
+                color: theme.sidebarHeaderTextColor,
+                disabledColor: changeOpacity(theme.sidebarHeaderTextColor, 0.32),
+                text: intl.formatMessage({id: 'edit_post.save', defaultMessage: 'Save'}),
+                enabled,
+            }],
+        });
+    }, [componentId, intl, theme]);
+
+    const onClose = useCallback(() => {
+        Keyboard.dismiss();
+        dismissModal({componentId});
+    }, [componentId]);
+
+    const handleUIUpdates = useCallback((res: {error?: unknown}) => {
+        if (res.error) {
+            setIsUpdating(false);
+            const errorMessage = intl.formatMessage({id: 'mobile.scheduled_post.update.error', defaultMessage: 'There was a problem editing this message. Please try again.'});
+            setErrorLine(errorMessage);
+        } else {
+            setIsUpdating(false);
+            onClose();
+        }
+    }, [intl, onClose]);
+
+    const onSavePostMessage = usePreventDoubleTap(useCallback(async () => {
+        setIsUpdating(true);
+        setErrorLine(undefined);
+        setErrorExtra(undefined);
+        toggleSaveButton(false);
+        if (!selectedTime.current) {
+            logDebug('ScheduledPostOptions', 'No time selected');
+            setIsUpdating(false);
+            const errorMessage = intl.formatMessage({id: 'mobile.scheduled_post.error', defaultMessage: 'No time selected'});
+            setErrorLine(errorMessage);
+            return;
+        }
+        const draftPayload = await draft.toApi(database);
```

Comment:
consider doing this at `updateSchedulePost` action instead of doing it here, you can allow the two types as the draftPayload argument and check the type, convert if needed.